### ISSUE TITLE
Possible false reporting of NS1001

### ISSUE THREAD

**Issue by manfred-brands** [nsubstitute/NSubstitute.Analyzers#148]
We have a Test class implementing an interface. The methods itself are not virtual. 

```
    public interface ITestInterface
    {
        void InterfaceMethod();
    }

    public class TestClass : ITestInterface
    {
        public void InterfaceMethod()
        {
        }
    }

    public sealed class TestClassFixture
    {
        [Test]
        public void TestNS1001()
        {
            TestClass instanceAsClass = Substitute.ForPartsOf<TestClass>();
            ITestInterface instanceAsInterface = Substitute.ForPartsOf<TestClass>();

            instanceAsClass.Received().InterfaceMethod();  // Triggers NS1001
            instanceAsInterface.Received().InterfaceMethod();
        }
    }
```

The code works either way and as the error message says: _Only interface members and virtual, overriding, and abstract members can be intercepted_. Either should be fine. Although the documentation says that members should be virtual.

The `ISymbolExtensions.IsInterfaceMember` checks on the `ContainingType` but it check if the method implements an interface.

If you agree, I will raise a PR.

**Comment by tpodolak** [comment]
The warning is not reported for
```c#
instanceAsInterface.Received().InterfaceMethod();
```
because NSubsitute.Analzyers are not yet smart enough to figure out that the substitute was created for class `Substitute.ForPartsOf<TestClass>();` ,please take a look [here](https://github.com/nsubstitute/NSubstitute.Analyzers/issues/68) for more detailed explanation. 

> The code works either way
This actually is not true, because in both cases you are getting false positives. Your test is green, but in fact it should be red, as method `InterfaceMethod` was never invoked

**Comment by manfred-brands** [comment]
Thanks, indeed whether actually called or not, the `Received` always passes.
